In [1]:
import pandas as pd
from mod06_deanonymize import load_data, link_records, deanonymization_rate

In [3]:
anon_path = "mod06_data/anonymized.csv"
aux_path = "mod06_data/auxiliary.csv"

### Load and review the data

In [4]:
anon, aux = load_data(anon_path, aux_path)

In [5]:
anon.head()

,anon_id,age,zip3,gender
0,ERM5C,50,432,F
1,OFJJP,75,323,F
2,I3H24,44,323,M
3,3WUYW,66,323,M
4,ZN91P,18,323,M


### Merge the datasets by age, gender, and zip3 and return the unique matches

In [6]:
matches = link_records(anon, aux)

### Calculate the match rate

In [7]:
rate = deanonymization_rate(matches, anon)
print(rate)

0.421


# Discussion Questions

### Describe your reaction to the match rate. What sources of data might be prone to privacy issues?

The match rate was 0.421, which is rather high considering the importance of keeping personal data private. Databases which store personal data concerning health, finance, etc. are sources of data prone to privacy; a single breach could expose a plethora of users, even if anonymized (as demonstrated above). Another example is social media; data surronding how to target ads for a certain user could be a big breach of privacy. There might also be concerns surronding smart technology and how they use and store data which is recorded audio or location.

### Provide examples of at least two other variables that could increase the match rate if included.

More precision on the location would increase match rate if included, since currently only the first 3 digits of a zipcode are availible, which leaves several users in the same area. Contact information such as phone numbers or emails would also help narrow who could be linked to a certain anon_id. Place of employment or place of education could also help increase match rate.